src/ProjectAudioManager.cpp

```diff
@@ -622,33 +623,17 @@ bool ProjectAudioManager::DoRecord(AudacityProject &project,
             if (prerollTrack)
                   transportTracks.prerollTracks.push_back(wt);
 
-            // A function that copies all the non-sample data between
-            // wave tracks; in case the track recorded to changes scale
-            // type (for instance), during the recording.
-            auto updater = [](Track &d, const Track &s){
-               auto &dst = static_cast<WaveTrack&>(d);
-               auto &src = static_cast<const WaveTrack&>(s);
-               dst.Reinit(src);
-            };
-
             // Get a copy of the track to be appended, to be pushed into
             // undo history only later.
             auto pending = std::static_pointer_cast<WaveTrack>(
                TrackList::Get( *p ).RegisterPendingChangedTrack(
-                  updater, wt.get() ) );
+                  nullptr, wt.get() ) );
 
             // End of current track is before or at recording start time.
             // Less than or equal, not just less than, to ensure a clip boundary.
             // when append recording.
             if (endTime <= t0) {
-
-               // Pad the recording track with silence, up to the
-               // maximum time.
-               auto newTrack = pending->EmptyCopy();
-               newTrack->InsertSilence(0.0, t0 - endTime);
-               newTrack->Flush();
-               pending->Clear(endTime, t0);
-               pending->Paste(endTime, newTrack.get());
```

Comment:
Only this change, and the changes in WaveTrack.cpp, are really addressing the issue as described.